### PR TITLE
! nested data casting in collections

### PR DESCRIPTION
**Pull request by pape-io**
Fixes nested data casting in case of collections.

### CODE REVIEW

**Comment by pape-io** [comment]
@cedricwider

File: lib/lhs/concerns/collection/internal_collection.rb [hunk 1]
```diff
@@ -33,8 +33,9 @@ def each(&_block)
       private
 
       def cast_item(item)
+        record = LHS::Record.for_url(item[:href]) if item[:href]
         data = LHS::Data.new(item, @parent, @record)
-        return @record.new(data) if @record
+        return (@record || record).new(data) if @record || record
```
Comment (acoiro) [inline lib/lhs/concerns/collection/internal_collection.rb]: quite hard to understand the difference between `@record` and `record`. Better naming would improve readability.

**Comment by acoiro** [comment]
otherwise 👍

File: lib/lhs/concerns/collection/internal_collection.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/concerns/collection/internal_collection.rb]: Giving better names actually made me realise that it should be the other way around... 🙈

**Comment by cedricwider** [comment]
👍

**Pull request merged**